core/src/main/scala/caliban/rendering/Renderer.scala

```diff
@@ -0,0 +1,559 @@
+            unsafeRenderObjectLike("type", description, name, interfaces, directives, fields, write)
+        }
+    }
+
+  private[caliban] lazy val interfaceTypeDefinitionRenderer: Renderer[InterfaceTypeDefinition] =
+    new Renderer[InterfaceTypeDefinition] {
+      override def unsafeRender(value: InterfaceTypeDefinition, indent: Option[Int], write: StringBuilder): Unit =
+        value match {
+          case InterfaceTypeDefinition(description, name, implements, directives, fields) =>
+            unsafeRenderObjectLike("interface", description, name, implements, directives, fields, write)
+        }
+    }
+
+  private def unsafeRenderObjectLike(
+    variant: String,
+    description: Option[String],
+    name: String,
+    implements: List[NamedType],
+    directives: List[Directive],
+    fields: List[FieldDefinition],
+    writer: StringBuilder
+  ): Unit = {
+    writer += '\n'
+    unsafeRenderDescription(description, None, writer)
+    writer ++= variant
+    writer += ' '
+    writer ++= name
+    implements match {
+      case Nil          =>
+      case head :: tail =>
+        writer ++= " implements "
+        writer ++= innerType(head)
+        tail.foreach { impl =>
+          writer ++= " & "
+          writer ++= innerType(impl)
+        }
+    }
+    directivesRenderer.unsafeRender(directives, None, writer)
+    writer += ' '
+    writer += '{'
+    fields.foreach { field =>
+      fieldDefinitionRenderer.unsafeRender(field, Some(1), writer)
+    }
+    writer += '\n'
+    writer += '}'
+    writer += '\n'
+  }
+
+  private lazy val directiveRenderer: Renderer[Directive] = new Renderer[Directive] {
+    override def unsafeRender(d: Directive, indent: Option[Int], writer: StringBuilder): Unit = {
+      writer ++= " @"
+      writer ++= d.name
+      var first = true
+      d.arguments.foreach { case (name, value) =>
+        if (first) {
+          writer += '('
+          writer ++= name
+          writer ++= ": "
+          writer ++= value.toInputString
+          first = false
+        } else {
+          writer ++= ", "
+          writer ++= name
+          writer ++= ": "
+          writer ++= value.toInputString
+        }
+      }
+      if (!first) writer += ')'
+    }
+  }
+
+  private lazy val directivesRenderer: Renderer[List[Directive]] =
+    directiveRenderer.list
+
+  private def unsafeRenderDescription(
+    description: Option[String],
+    indent: Option[Int],
+    writer: StringBuilder,
+    newline: Boolean = true
+  ): Unit = {
+    val tripleQuote = "\"\"\""
+
+    def nlOrSp  = if (newline) writer += '\n' else writer += ' '
+    def nlOrNot = if (newline) writer += '\n' else writer
+
+    description.foreach {
+      case value if value.contains('\n') =>
+        val valueEscaped = value.replace(tripleQuote, s"\\$tripleQuote")
+        // check if it ends in quote but it is already escaped
+        if (value.endsWith("\\\"")) {
+          writer ++= tripleQuote
+          nlOrNot
```

Comment:
I had to stare at this for the better part of a minute before realising it's a side-effecting method 😅. Maybe rename them to  `nlOrSp()` and `nlOrNot()` to make it more readable?

Reply:
Ok will fix